File: 1-two-sum/two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        map<int,int>m;
        int n = nums.size();
        for(int i = 0; i<n;i++)
        {
            int check = target - nums[i];
            if(m.find(check) != m.end())
            {
                return {m[check], i};
            }
            else
            {
                m[nums[i]] = i;
            }
        }
        return {-1,-1};
    }
};

File: 1-two-sum/two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        n = len(nums)
        s = {}
        for i in range(n):
            needed = target - nums[i]
            if needed in s:
                return [s[needed], i]
            s[nums[i]] =  i 
        return [-1, -1]